google/cloud/forseti/common/opencensus/tracing.py

```diff
@@ -0,0 +1,347 @@
+# Copyright 2017 The Forseti Security Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Forseti OpenCensus gRPC tracing setup."""
```

Comment:
The code in this module is pretty intricate.  Are they based on any resources or documentations that I can reference or look at?

Reply:
This is the actual tracing code from distributed tracing framework - OpenCensus.
https://github.com/census-instrumentation/opencensus-python

Reply:
Can you please point me to the where the `outer_wrapper`, `inner_wrapper` pattern is documented or re-implemented from?  My ultimate question is if these code are necessary to add to our codebase.

Reply:
@blueandgold this pattern (see [here](https://stackoverflow.com/a/5929165)) allows to pass arguments to our decorators. It's useful when we want to get an OpenCensus tracer from a class / method attribute, and should probably be implemented in OpenCensus itself later on.